### ISSUE TITLE
can't update traffic manager profiles

### ISSUE THREAD

**Issue by Katana-Steel** [Azure/azure-sdk-for-python#1562]
I'm trying to update the DNS TTL from az, however I keep getting this error:

> Endpoint name '' is invalid. Endpoint names cannot be empty or all whitespace, and must be up to 260 alphanumeric characters long. The type '<null>' specified for endpoint '<null>' is invalid. Valid endpoint types are: 'Microsoft.Network/trafficManagerProfiles/azureEndpoints', 'Microsoft.Network/trafficManagerProfiles/externalEndpoints', 'Microsoft.Network/trafficManagerProfiles/nestedEndpoints'. There are duplicates in the list of endpoints. The following endpoint(s) appear more than once: .

the debug output that indeed it adds to the endpoint list to the PUT request... however according to the example: [Profiles - Update (Traffic Manager) | Microsoft Docs](https://docs.microsoft.com/en-us/rest/api/trafficmanager/profiles/update#Examples) 
only the sections that is being updated needs to be present in the PUT request

this is the request body for my simple update
`az --debug network traffic-manager profile update -g <resG> -n <tm_profile> --ttl 60`
```
{
    "location": "global", 
    "properties": {
        "profileStatus": "Enabled", 
        "endpoints": [
            {
                "properties": {
                    "endpointMonitorStatus": "Online",
                    "priority": 1,
                    "targetResourceId": "<my_id_for_the_first_endpoint's_target>",
                    "target": "<first_target's_fqdn>",
                    "weight": 1,
                    "endpointLocation": "A",
                    "endpointStatus": "Enabled"
                }
            },
            {
                "properties": {
                    "endpointMonitorStatus": "Disabled",
                    "priority": 4,
                    "targetResourceId": "<my_id_for_the_second_endpoint's_target>",
                    "target": "<second_target's_fqdn>",
                    "weight": 1,
                    "endpointLocation": "B",
                    "endpointStatus": "Disabled"
                }
            }
        ],
        "trafficRoutingMethod": "Priority",
        "monitorConfig": {"port": 80, "protocol": "TCP", "intervalInSeconds": 30, "toleratedNumberOfFailures": 3, "profileMonitorStatus": "Online", "timeoutInSeconds": 10},
        "dnsConfig": {"relativeName": "<TM_dns_name>", "ttl": 60}
    },
    "tags": {}
}
```

**Comment by lmazuel** [comment]
Hi @Katana-Steel , please submit this issue on the CLI side:
https://github.com/Azure/azure-cli/issues

I keep this one opened until we are sure it's CLI and not SDK. FYI @tjprescott

**Comment by Katana-Steel** [comment]
thank you, hopefully we can get this resolved soon. 
I'm creating script to update ttl, enabling/disabling endpoints, and endpoint priority, 
to move client traffic on multiple TMs between regions.

**Comment by Katana-Steel** [comment]
fixed since Azure-cli release 2.0.21 
thank you
